workbench-core/datasets-ui/src/utils.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { nameRegex, ValidationRule } from '@aws/workbench-core-swb-common-ui';
+
+export const datasetNameValidationRules: ValidationRule<string | undefined>[] = [
+  {
+    condition: (a: string | undefined) => !!a,
```

Comment:
maybe descriptive variable names instead of `a` to help clarify what the item being validated is?